pytential/symbolic/primitives.py

```diff
@@ -1166,9 +1224,26 @@ def __post_init__(self) -> None:
 
 
 def interp(from_dd, to_dd, operand):
+    warn("Calling 'interp' is deprecated and it will be removed in 2025. Use "
+         "'interpolate' instead (has a different argument order).",
+         DeprecationWarning, stacklevel=2)
+
     return Interpolation(as_dofdesc(from_dd), as_dofdesc(to_dd), operand)
 
 
+@for_each_expression
+def interpolate(operand: ExpressionT,
+                from_dd: DOFDescriptorLike,
+                to_dd: DOFDescriptorLike) -> ExpressionT | Interpolation:
+    from_dd = as_dofdesc(from_dd)
+    to_dd = as_dofdesc(to_dd)
+
+    if from_dd == to_dd:
+        return operand
+
+    return Interpolation(from_dd, to_dd, operand)
```

Comment:
Is this ok? Introduced the new one just because the argument order didn't work :\

Reply:
Hmm. I generally prefer having the "how" parameters before the "what" ones. At the same time, there is precedent for "what-before-how" (`ElementwiseMin`), so what do I know? It seems that this at least beats some consistency into things, so I'm OK.

Reply:
@inducer Yeah, I remembered we did the `interp` one in that order on purpose. 

Some ways to make that work would be to 
* make `for_each_expression` work like a multimap, so that it would vectorize over all arrays in `*args`.
* make `for_each_expression(vec_args=n)` to be able to tell it which argument is meant to be vectorized.

Does any of that sound good?

Reply:
I'm not sure any of those could be statically typed.

Reply:
Hm, yeah, fair point, can't think of a way to do that.

For comparison, this is what `jax.jit` is doing (where e.g. `static_argnums` works in a similar way to my second point):
https://github.com/jax-ml/jax/blob/b372ce4b1ab0bee7a1da495b098ff3948a6c0d4d/jax/_src/api.py#L141=L155

Reply:
The new `interpolate` can be typed straightforwardly, so I don't see a good reason to deviate from it.